src/Task/Stylelint.php

```diff
@@ -0,0 +1,140 @@
+            'whitelist_patterns' => null,
+
+            // Stylelint native config options
+            'config' => null,
+            'config_basedir' => null,
+            'ignore_path' => null,
+            'ignore_pattern' => null,
+            'syntax' => null,
+            'custom_syntax' => null,
+            'ignore_disables' => null,
+            'disable_default_ignores' => null,
+            'cache' => null,
+            'cache_location' => null,
+            'formatter' => null,
+            'custom_formatter' => null,
+            'quiet' => null,
+            'color' => null,
+            'report_needless_disables' => null,
+            'report_invalid_scope_disables' => null,
+            'report_descriptionless_disables' => null,
+            'max_warnings' => null,
+            'output_file' => null,
+        ]);
+
+        // Task config options
+        $resolver->addAllowedTypes('bin', ['null', 'string']);
+        $resolver->addAllowedTypes('triggered_by', ['array']);
+        $resolver->addAllowedTypes('whitelist_patterns', ['null', 'array']);
+
+        // Stylelint native config options
+        $resolver->addAllowedTypes('config', ['null', 'string']);
+        $resolver->addAllowedTypes('config_basedir', ['null', 'string']);
+        $resolver->addAllowedTypes('ignore_path', ['null', 'string']);
+        $resolver->addAllowedTypes('ignore_pattern', ['null', 'string']);
+        $resolver->addAllowedTypes('syntax', ['null', 'string']);
+        $resolver->addAllowedTypes('custom_syntax', ['null', 'string']);
+        $resolver->addAllowedTypes('ignore_disables', ['null', 'bool']);
+        $resolver->addAllowedTypes('disable_default_ignores', ['null', 'bool']);
+        $resolver->addAllowedTypes('cache', ['null', 'bool']);
+        $resolver->addAllowedTypes('cache_location', ['null', 'string']);
+        $resolver->addAllowedTypes('formatter', ['null', 'string']);
+        $resolver->addAllowedTypes('custom_formatter', ['null', 'string']);
+        $resolver->addAllowedTypes('quiet', ['null', 'bool']);
+        $resolver->addAllowedTypes('color', ['null', 'bool']);
+        $resolver->addAllowedTypes('report_needless_disables', ['null', 'bool']);
+        $resolver->addAllowedTypes('report_invalid_scope_disables', ['null', 'bool']);
+        $resolver->addAllowedTypes('report_descriptionless_disables', ['null', 'bool']);
+        $resolver->addAllowedTypes('max_warnings', ['null', 'integer']);
+        $resolver->addAllowedTypes('output_file', ['null', 'string']);
+
+        return $resolver;
+    }
+
+    public function canRunInContext(ContextInterface $context): bool
+    {
+        return ($context instanceof GitPreCommitContext || $context instanceof RunContext);
+    }
+
+    public function run(ContextInterface $context): TaskResultInterface
+    {
+        $config = $this->getConfig()->getOptions();
+
+        $files = $context
+            ->getFiles()
+            ->paths($config['whitelist_patterns'] ?? [])
```

Comment:
maybe better to always make it an array instead of nullable

Reply:
should I update the ESlint task too? I think it's a BC break for those who use `null` already

Reply:
For eslint this might a BC break indeed.
We could add a deprecation warning when using the `null` value there:
https://symfony.com/doc/current/components/options_resolver.html#deprecating-the-option

But that seems out of scope for this PR.

Reply:
I created https://github.com/phpro/grumphp/issues/911 so this is not forgotten